elyra/pipeline/pipeline.py

```diff
@@ -440,3 +406,72 @@ def __eq__(self, other: "Pipeline") -> bool:
                 and self.runtime_config == other.runtime_config
                 and self.operations == other.operations
             )
+
+
+class KeyValueList(list):
+    """
+    A list class that exposes functionality specific to lists whose entries are
+    key-value pairs separated by a pre-defined character.
+    """
+
+    _key_value_separator: str = "="
+
+    def to_dict(self, logger: Optional[Logger] = None) -> Dict[str, str]:
+        """
+        Properties consisting of key/value pairs are stored in a list of separated
+        strings, while most processing steps require a dictionary - so we must convert.
+        If no key/value pairs are specified, an empty dictionary is returned, otherwise
+        pairs are converted to dictionary entries and returned.
+        """
+        kv_dict = {}
+        for kv in self:
+            if not kv:
+                continue
+
+            if self._key_value_separator not in kv:
+                raise ValueError(
+                    f"Property {kv} does not contain the expected "
+                    f"separator character: '{self._key_value_separator}'."
+                )
+
+            key, value = kv.split(self._key_value_separator, 1)
+            if not key.strip():
+                KeyValueList.log_message(f"Skipping inclusion of property '{kv}': no key found", logger, logging.WARN)
+                continue
+            if not value:
+                KeyValueList.log_message(
+                    f"Skipping inclusion of property '{key}': no value specified", logger, logging.INFO
+                )
+                continue
+
+            kv_dict[key] = value
+        return kv_dict
+
+    def dict_to_kv_list(self, kv_dict: Dict) -> "KeyValueList":
+        """
+        Convert a set of key-value pairs stored in a dictionary to
+        a KeyValueList of strings with the defined separator.
+        """
+        str_list = [f"{key}{self._key_value_separator}{value}" for key, value in kv_dict.items()]
+        return KeyValueList(str_list)
+
+    def merge_kv_pairs(self, primary_list: "KeyValueList") -> "KeyValueList":
+        """
+        Merge two key-value pair lists, preferring the values given in the
+        primary_list in the case of a matching key between the two lists.
+        """
+        primary_dict = primary_list.to_dict()
+        secondary_dict = self.to_dict()
+
+        merged_list = self.dict_to_kv_list({**secondary_dict, **primary_dict})
+        return KeyValueList(merged_list)
```

Comment:
I find it more intuitive if this action is performed against the instance on which the method is called.  In this case, I think it would be the current instance is the "primary", so an instance is basically being merged with a "secondary".   I'm just not sure how to "adjust" self to the new list.

Perhaps it would be cleaner to also let this be a class factory pattern as well, using another class method:
```suggestion
    @classmethod
    def merge(self, primary: "KeyValueList", secondary: "KeyValueList) -> "KeyValueList":
        """
        Merge two key-value pair lists, preferring the values given in the
        primary parameter in the case of a matching key between the two lists.
        """
        primary_dict = primary.to_dict()
        secondary_dict = secondray.to_dict()
        return KeyValueList.from_dict({**secondary_dict, **primary_dict})
```

Reply:
That would make more sense. And agreed on the primary vs secondary meaning.